virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,188 @@
+    :param params: parameters to load with, e.g. 'key1=param1 key2=param2 ...'
+    :return: instance if module was loaded
+    """
+
+    if params != "" or force:
+        handler = KernelModuleHandler(module_name)
+        handler.reload_module(force, params)
+        return handler
+    return None
+
+
+class KernelModuleHandler(object):
+    """Class handling Linux kernel modules"""
+
+    def __init__(self, module_name):
+        """Create kernel module handler"""
+
+        self._module_name = module_name
+        self._was_loaded = None
+        self._loaded_config = None
+        self._config_backup = None
+        self._backup_config()
+
+    def reload_module(self, force, params):
+        """
+        Reload module with given parameters.
+
+        If force=False loading will be skipped when either the module is already loaded
+        with the passed parameters or when the module has not been loaded at all.
+
+        +-----------------------+-+-+-+-+-+-+
+        |**precondition**                   |
+        +-----------------------+-+-+-+-+-+-+
+        |module loaded          |N|N|Y|Y|Y|Y|
+        +-----------------------+-+-+-+-+-+-+
+        |params already loaded  |*|*|Y|N|Y|N|
+        +-----------------------+-+-+-+-+-+-+
+        |force load             |Y|N|Y|Y|N|N|
+        +-----------------------+-+-+-+-+-+-+
+        |**result**                         |
+        +-----------------------+-+-+-+-+-+-+
+        |load module with params|Y|N|Y|Y|N|Y|
+        +-----------------------+-+-+-+-+-+-+
+
+        :param force: if to force load with params in any case, e.g. True
+        :param params: parameters to load with, e.g. 'key1=param1 key2=param2 ...'
+        """
+
+        current_config = self._get_serialized_config()
+        if not force:
+            do_not_load = False
+            if current_config and all(x in current_config.split() for x in params.split()):
+                logging.debug("Not reloading module. Current module configuration for %s"
+                              " already contains all requested parameters."
+                              " Requested: '%s'. Current: '%s'. Use force=True to force"
+                              " loading.", self._module_name, params, current_config)
+                do_not_load = True
+            elif not self._was_loaded:
+                logging.debug("Module %s isn't loaded. Use force=True to force"
+                              " loading.", self._module_name)
+                do_not_load = True
+            if do_not_load:
+                return
+
+        cmd = ""
+        if self._was_loaded:
+            cmd += 'rmmod %s; ' % self._module_name
+        cmd += 'modprobe %s %s' % (self._module_name, params)
+        cmd = cmd.strip()
+        logging.debug("Loading module: %s", cmd)
+        status, output = process.getstatusoutput(cmd, shell=True,
+                                                 ignore_status=True)
```

Comment:
allowing `shell=True` can lead to unexpected escapes or security threads. Yes, we are testing framework, using it extensively, but we should not do that without a reason. Would you please use `shell=False` and issue the `rmmod` separately i `was_loaded`?